caveclient/skeletonservice.py

```diff
@@ -0,0 +1,164 @@
+    auth_client=None,
+    api_version="latest",
+    version=None,
+    verify=True,
+    max_retries=None,
+    pool_maxsize=None,
+    pool_block=None,
+    over_client=None,
+) -> "SkeletonClientV1":
+    """Factory for returning SkeletonClient
+
+    Parameters
+    ----------
+    server_address : str
+        server_address to use to connect to (i.e. https://minniev1.microns-daf.com)
+    datastack_name : str
+        Name of the datastack.
+    auth_client : AuthClient or None, optional
+        Authentication client to use to connect to server. If None, do not use authentication.
+    api_version : str or int (default: latest)
+        What version of the api to use, 0: Legacy client (i.e www.dynamicannotationframework.com)
+        2: new api version, (i.e. minniev1.microns-daf.com)
+        'latest': default to the most recent (current 2)
+    version : default version to query
+        if None will default to latest version
+
+    Returns
+    -------
+    ClientBaseWithDatastack
+        List of datastack names for available datastacks on the annotation engine
+    """
+
+    if auth_client is None:
+        auth_client = AuthClient()
+
+    auth_header = auth_client.request_header
+    endpoints, api_version = _api_endpoints(
+        api_version,
+        SERVER_KEY,
+        server_address,
+        skeletonservice_common,
+        skeletonservice_api_versions,
+        auth_header,
+        fallback_version=1,
+        verify=verify,
+    )
+
+    SkeletonClient = client_mapping[api_version]
+    return SkeletonClient(
+        server_address,
+        auth_header,
+        api_version,
+        endpoints,
+        SERVER_KEY,
+        max_retries=max_retries,
+        pool_maxsize=pool_maxsize,
+        pool_block=pool_block,
+        over_client=over_client,
+    )
+
+class SkeletonClientV1(ClientBase):
+    def __init__(
+        self,
+        server_address,
+        auth_header,
+        api_version,
+        endpoints,
+        server_name,
+        max_retries=None,
+        pool_maxsize=None,
+        pool_block=None,
+        over_client=None,
+    ):
+        super(SkeletonClientV1, self).__init__(
+            server_address,
+            auth_header,
+            api_version,
+            endpoints,
+            server_name,
+            max_retries=max_retries,
+            pool_maxsize=pool_maxsize,
+            pool_block=pool_block,
+            over_client=over_client,
+        )
+    
+    def get_skeleton_info(
+        self,
+        datastack_name,
+    ):
+        endpoint_mapping = self.default_url_mapping
+        endpoint_mapping["datastack_name"] = datastack_name
+
+        url = self._endpoints["skeleton_info"].format_map(endpoint_mapping)
+        print(f"get_skeleton_info() url: {url}")
+
+        response = self.session.get(url)
+        return handle_response(response, False)
+    
+    def get_skeleton_by_datastack_and_rid(
+        self,
+        datastack_name,
+        rid,
+    ):
```

Comment:
what do people think about just `get_skeleton`? I would also advocate for the root being the first argument, and calling it `root_id`. I think this is just more intuitive and matches existing naming better.

Reply:
and i guess i'd also expect the datastack to be None by default, and inherit from the client when not set

Reply:
Agreed 100%. Also, `datastack_name` should be optional and None by default, in which case it uses the datastack set on instance creation.

```python
  def get_skeleton_by_datastack_and_rid(
         self,
         root_id: int,
         datastack_name: Optional[str] = None,
     ):
```

Reply:
I don't think I understand why the options in the other function can't be integrated into a single `get_skeleton` function with all of these options and sensible defaults.

Reply:
I specified the details because there are multiple functions with varying arguments. I guess they can be folded into a single function with optional arguments instead.

Reply:
Re: Optional datastack_name. Noted. Thanks.